bindings/go/configuration/extract/v1alpha1/spec/config.go

```diff
@@ -0,0 +1,82 @@
+package spec
+
+import (
+	"fmt"
+
+	v1 "ocm.software/open-component-model/bindings/go/configuration/generic/v1/spec"
+	"ocm.software/open-component-model/bindings/go/runtime"
+)
+
+const (
+	// ConfigType defines the type identifier for transformation configurations
+	ConfigType = "extract.oci.artifact.ocm.software"
+)
+
+var scheme = runtime.NewScheme()
+
+func init() {
+	scheme.MustRegisterWithAlias(&Config{}, runtime.NewVersionedType(ConfigType, Version))
+}
+
+// Layer has configuration information for layers to process.
+type Layer struct {
+	Index     string `json:"index,omitempty"`
+	MediaType string `json:"mediaType,omitempty"`
+}
+
+// Config represents the top-level configuration for the transformation.
+//
+// +k8s:deepcopy-gen:interfaces=ocm.software/open-component-model/bindings/go/runtime.Typed
+// +k8s:deepcopy-gen=true
+// +ocm:typegen=true
+type Config struct {
+	Type runtime.Type `json:"type"`
+	// Layer defines configuration options on selecting the right Layer.
+	Layer Layer `json:"layer,omitempty"`
```

Comment:
Generally this looks much better. Im just wondering why the media type is needed? is this a selector? How is this supposed to work. I was missing the logic for the behavior on the structs

Reply:
yes! as per specification you need to identify the correct layer by media type. but if we want a selector the spec should be a proper selector imo

Reply:
I think we should invest in a proper specification here. That means for me that we need to find out what we want.

I think we want to be able to somehow tell the transformer to select _multiple_ layers by either _index_ or _mediaType_ or in the future even _annotations_ or other fields. That means we should probably invest in a proper selector here.

Additionally we need to identify how to map this layer to the filesystem. I am open for suggestions here

Reply:
Okay, I added a basic selection logic. I feel like I might have went to far with it though...

Reply:
I felt like, if we _do_ provide a selector, we actually also have to provide a nice way of matching. So I added LayerInfo and a properties type of matcher.